core/capabilities/triggers/logevent/trigger.go

```diff
@@ -0,0 +1,207 @@
+// in a loop with a periodic delay of pollPeriod milliseconds, which is specified in
+// the job spec
+type logEventTrigger struct {
+	ch   chan<- capabilities.TriggerResponse
+	lggr logger.Logger
+
+	// Contract address and Event Signature to monitor for
+	reqConfig      *RequestConfig
+	contractReader types.ContractReader
+	relayer        core.Relayer
+	startBlockNum  uint64
+
+	// Log Event Trigger config with pollPeriod and lookbackBlocks
+	logEventConfig Config
+	ticker         *time.Ticker
+	done           chan bool
+	wg             sync.WaitGroup
+}
+
+// Construct for logEventTrigger struct
+func newLogEventTrigger(ctx context.Context,
+	reqConfig *RequestConfig,
+	logEventConfig Config,
+	relayer core.Relayer) (*logEventTrigger, chan capabilities.TriggerResponse, error) {
+	jsonBytes, err := json.Marshal(reqConfig.ContractReaderConfig)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Create a New Contract Reader client, which brings a corresponding ContractReader gRPC service
+	// in Chainlink Core service
+	contractReader, err := relayer.NewContractReader(ctx, jsonBytes)
+	if err != nil {
+		return nil, nil,
+			fmt.Errorf("error fetching contractReader for chainID %d from relayerSet: %v", logEventConfig.ChainID, err)
+	}
+
+	// Bind Contract in ContractReader
+	boundContracts := []types.BoundContract{{Name: reqConfig.ContractName, Address: reqConfig.ContractAddress}}
+	err = contractReader.Bind(ctx, boundContracts)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Get current block HEAD/tip of the blockchain to start polling from
+	latestHead, err := relayer.LatestHead(ctx)
+	if err != nil {
+		return nil, nil, fmt.Errorf("error getting latestHead from relayer client: %v", err)
+	}
+	height, err := strconv.ParseUint(latestHead.Height, 10, 64)
+	if err != nil {
+		return nil, nil, fmt.Errorf("invalid height in latestHead from relayer client: %v", err)
+	}
+	startBlockNum := uint64(0)
+	if height > logEventConfig.LookbackBlocks {
+		startBlockNum = height - logEventConfig.LookbackBlocks
+	}
+
+	// Setup callback channel, logger and ticker to poll ContractReader
+	callbackCh := make(chan capabilities.TriggerResponse, defaultSendChannelBufferSize)
+	pollPeriod := int64(1000) // default pollPeriod is 1s
+	if logEventConfig.PollPeriod <= math.MaxInt64 {
+		pollPeriod = int64(logEventConfig.PollPeriod)
+	}
+	ticker := time.NewTicker(time.Duration(pollPeriod) * time.Millisecond)
+	done := make(chan bool)
+	lggr, err := logger.New()
+	if err != nil {
+		return nil, nil, fmt.Errorf("could not initialise logger for LogEventTrigger")
+	}
+
+	// Initialise a Log Event Trigger
+	l := &logEventTrigger{
+		ch:   callbackCh,
+		lggr: logger.Named(lggr, "LogEventTrigger: "),
+
+		reqConfig:      reqConfig,
+		contractReader: contractReader,
+		relayer:        relayer,
+		startBlockNum:  startBlockNum,
+
+		logEventConfig: logEventConfig,
+		ticker:         ticker,
+		done:           done,
+	}
+	go l.Listen(ctx)
```

Comment:
@kidambisrinivas We need to ensure that this is tracked, and probably we don't want to obey the cancellation of the parent context, so you'll want to use context.WithoutCancel(ctx) here

Reply:
The Listen function is tracked with a separate done channel, to which a message is sent when Stop function is called from the Service's Close function. The parent ctx is sent to QueryKey function.

Reply:
Fixed this